Controller/ActionController.php

```diff
@@ -0,0 +1,208 @@
+                  $_SESSION['lname'],
+                  $_SESSION['email'],
+                  password_hash($_SESSION['password'], PASSWORD_DEFAULT)
+                ]
+              )
+            ) {
+              $msg = 'Successfully Inserted Try Logging In';
+              $cls = 'green';
+              $db_obj->closeDb();
+            }
+            else {
+              $msg = 'Could not save data';
+            }
+          }
+        }
+      }
+    }
+    return [$msg,$cls];
+  }
+
+  /**
+   * Function to handle reset form actions.
+   *
+   * @return array
+   *   Returns an array of size 2 where first element is the status
+   *   message and second is the css class.
+   */
+  public function resetController() {
+
+    $msg = '';
+    $cls = 'red';
+
+    if (isset($_POST['submit'])) {
+      if (empty($_POST['otp']) || !preg_match("/^\d{4}$/", $_POST['otp'])) {
+        $msg = 'Invalid OTP';
+      }
+      else {
+        if ($_POST['otp'] != $_SESSION['otp']) {
+          $msg = 'OTP Does Not Match.';
+        }
+        else {
+          if (strtotime('now') > $_SESSION['valid_time']) {
+            $msg = 'OTP Expired';
+            header('location:/');
+          }
+          else {
+            $valid = new Validator();
+            if ($valid->isValidPassword($_POST['new_password']) && ($_POST['new_password'] == $_POST['confirm_password'])) {
+              $db_obj =
+              new Database($_ENV['HOST_NAME'], $_ENV['DB_NAME'], $_ENV['USER_NAME'], $_ENV['DB_PASSWORD']);
+              if ($db_obj->updateInto(
+                'user',
+                ['password'],
+                [password_hash($_POST['new_password'],PASSWORD_DEFAULT)],
+                $_SESSION['email'])) {
+                  $msg = 'Successfully Updated Try Logging In';
+                  $cls = 'green';
+                  $db_obj->closeDb();
+              }
+              else {
+                $msg = "Data could not be saved, Try again.";
+              }
+            }
+            else {
+              $msg = 'Check password fields!';
+            }
+          }
+        }
+      }
+    }
+    return [$msg,$cls];
+  }
+
+  /**
+   * Function to set variables on homepage loading.
+   *
+   * @return array
+   *   Returns the full name and the image source of the user.
+   */
+  public function homeController() {
+    $email = $_SESSION['email'];
+    $db_obj = new Database($_ENV['HOST_NAME'], $_ENV['DB_NAME'], $_ENV['USER_NAME'], $_ENV['DB_PASSWORD']);
+    $res = $db_obj->selectUser('user', $email);
+    return [$res['first_name']. ' ' . $res['last_name'], $res['image']];
+  }
+
+  /**
+   * Function to set variable in profile section.
+   *
+   * @return array
+   *   Returns the data to be shown in profile section,
+   *   like image, name, and email.
+   */
+  public function profileController() {
+    $email = $_SESSION['email'];
+    $db_obj = new Database($_ENV['HOST_NAME'], $_ENV['DB_NAME'], $_ENV['USER_NAME'], $_ENV['DB_PASSWORD']);
```

Comment:
This is repeated multple times in this file. 
Can we maybe create a helper function and invoke it whenever required